jax/experimental/jax2tf/tests/tf_test_util.py

```diff
@@ -111,12 +128,23 @@ def run_tf(mode):
       else:
         assert False
 
+    def is_tf_exception(lim: jax2tf.jax2tf.Limitation):
+      return lim.ErrorType == 'Missing TF support'
+
     result_tf = None
     for mode in ("eager", "graph", "compiled"):
+      current_limitations = jax2tf.jax2tf.all_limitations[:]
       try:
         result_tf = run_tf(mode)
       except Exception as e:
-        if not expect_tf_exceptions:
+        detected_tf_exception = False
+
+        if generate_tests_from_categorize:
+          new_limitations = (
+            jax2tf.jax2tf.all_limitations[len(current_limitations):])
+          detected_tf_exception = any(map(is_tf_exception, new_limitations))
+
+        if not (expect_tf_exceptions or detected_tf_exception):
           raise e
         else:
           print(f"Encountered expected exception for mode={mode}: {e}")
```

Comment:
Can we print here the messages from the relevant limitations?

Reply:
Done in `4cb23e6`